### PR TITLE
Pages/product listing

### PR DESCRIPTION
**Pull request by manish591**
### Added product listing page.

### CODE REVIEW

File: css/product-listing.css [hunk 1]
```diff
@@ -0,0 +1,366 @@
+/* Main container */
+
+.listing {
+  margin-block-start: 5rem;
+}
+
+.listing__wrapper {
+  grid-template-columns: 1fr;
+  inline-size: min(90%, 1200px);
```
Comment (tejas-muthya37) [inline css/product-listing.css:9]: Try 75 rem instead of 1200px for maintaining consistency in units.

File: css/product-listing.css [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+.card {
+  inline-size: unset;
+}
+
+/* listing products */
+
+.listing__products {
+  gap: 2rem;
+  --grid-auto-item-size: 195px;
```
Comment (tejas-muthya37) [inline css/product-listing.css:97]: Can use rem here. 12.25 rem would be close.

File: css/product-listing.css [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+.card__realPrice {
+  font-size: 0.7rem;
+  text-decoration: line-through;
+}
+
+.card__discount {
+  font-size: 0.8rem;
+  font-weight: 700;
+  color: #734124;
```
Comment (tejas-muthya37) [inline css/product-listing.css]: Try using variables for colors. It makes the job of maintaining and reviewing code so much easier :)

File: css/product-listing.css (hunk 1 above)
Comment (vinc3nati) [inline css/product-listing.css:9]: nice use of `min()`

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const listingFilter = document.querySelector('.listing__filters');
+const btn = document.querySelector('.listing-header__right');
+
+
+btn.addEventListener('click', e => {
+    if(listingFilter.classList.contains('filter-list__open')) {
```
Comment (tejas-muthya37) [inline js/script.js]: The act of removing a class if it is already present and adding the class if not can be achieved by classList.toggle(<class-name>)

Makes the code much more crisp.

File: pages/product-listing.html [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+              </div>
+                <form class="listing__form">
+                    <select multiple="true" class="filters__select" title="Filter results by category">
+                        <optgroup label="Availabilty">
+                            <option value="Include out of stock">Include out of stock</option>
+                            <option value="Fast delivery only">Fast delivery only</option>
+                        </optgroup>
+                        <optgroup label="Prices">
+                            <option value="Rs. 200 to Rs. 1000">Rs. 200 to Rs. 1000</option>
+                            <option value="Rs. 1000 to Rs. 5000">Rs. 1000 to Rs 5000</opt10on>
```
Comment (tejas-muthya37) [inline pages/product-listing.html]: The closing tag has a typo. (opt10on)

**Review by vinc3nati — APPROVED** [review]
Looks good.

**Review by tejas-muthya37 — APPROVED** [review]
Rest looks good to merge. Approving this PR. Good luck :)

**Pull request merged**